tests/unit/models/test_projects.py

```diff
@@ -158,12 +158,29 @@ def test_app_defaults(self, project_yaml_data):
 class TestProjectValidation:
     """Validate top-level project items."""
 
+    def test_reentrant_validation(self, project_yaml_data):
+        """Validators should be reentrant.
+
+        When changing a field, validators should be able to validate existing data.
+        For example, a "before" validator will see "build-base" when creating the model
+        and "build_base" when re-validating the model.
+        """
+        data = project_yaml_data(
+            base="bare",
+            # build-base has to be parsed for the validator to allow 'architectures'
+            build_base="core22",
+            architectures=["amd64"],
+        )
+        project = Project.unmarshal(data)
+
+        project.version = cast(VersionStr, "1.2.3")
```

Comment:
what is the purpose of setting this? doesn't the test end right after?

Reply:
I re-worded the test name and comments to categorize this as a specific regression test.

The short of it is that changing a value in a model will re-run model validators.  So this tests that an arbitrary value in the model can be updated without causing an unexpected validation error on reading `build-base`.